index.test.js

```diff
@@ -124,3 +131,225 @@ test('shows lines that are missing coverage when failure occurs', () => {
+    id: 2,
+    user: { type: 'Bot' },
+    body: 'some comment ' + signature,
+  };
+  const comments = { data: [comment1, comment2] };
+  const octokit = {
+    rest: {
+      issues: {
+        listComments: (_) => comments,
+      },
+    },
+  };
+  github.getOctokit = (_) => octokit;
+  const context = {
+    repo: {},
+    issue: { number: 1 },
+  };
+  github.context = context;
+
+  const result = await getSignedBotCommentIdentifier('token', signature);
+  expect(result).toBe(comment2.id);
+});
+
+test('updateGitHubComment does nothing when token is null', async () => {
+  const updateComment = jest.fn((_) => {});
+  const octokit = {
+    rest: {
+      issues: {
+        updateComment: updateComment,
+      },
+    },
+  };
+  github.getOctokit = (_) => octokit;
+
+  await updateGitHubComment(null, 1, 'comment');
+  expect(updateComment).not.toHaveBeenCalled();
+});
+
+test('updateGitHubComment does nothing when identifier is null', async () => {
+  const updateComment = jest.fn((_) => {});
+  const octokit = {
+    rest: {
+      issues: {
+        updateComment: updateComment,
+      },
+    },
+  };
+  github.getOctokit = (_) => octokit;
+
+  await updateGitHubComment('token', null, 'comment');
+  expect(updateComment).not.toHaveBeenCalled();
+});
+
+test('updateGitHubComment is called with correct data', async () => {
+  const updateComment = jest.fn((_) => {});
+
+  const octokit = {
+    rest: {
+      issues: {
+        updateComment: updateComment,
+      },
+    },
+  };
+  github.getOctokit = (_) => octokit;
+  const context = {
+    repo: {},
+    issue: { number: 1 },
+  };
+  github.context = context;
+
+  const comment = {
+    id: 1,
+    body: 'some comment',
+  };
+  await updateGitHubComment('token', comment.id, comment.body);
+  expect(updateComment).toHaveBeenCalledWith({
+    ...context.repo,
+    issue_number: context.issue.number,
+    comment_id: comment.id,
+    body: comment.body,
+  });
+});
+
+test('createGitHubComment does nothing when token is null', async () => {
+  const createComment = jest.fn((_) => {});
+  const octokit = {
+    rest: {
+      issues: {
+        createComment: createComment,
+      },
+    },
+  };
+  github.getOctokit = (_) => octokit;
+
+  await createGitHubComment(null, 1, 'comment');
+  expect(createComment).not.toHaveBeenCalled();
+});
+
+test('createGitHubComment is called with correct data', async () => {
+  const createComment = jest.fn((_) => {});
+  const octokit = {
+    rest: {
+      issues: {
+        createComment: createComment,
+      },
+    },
+  };
+  github.getOctokit = (_) => octokit;
+  const context = {
+    repo: {},
+    issue: { number: 1 },
+  };
+  github.context = context;
+
+  const comment = 'some comment';
+  await createGitHubComment('token', comment);
+  expect(createComment).toHaveBeenCalledWith({
+    ...context.repo,
+    issue_number: context.issue.number,
+    body: comment,
+  });
+});
+
+test('formatCoverageAsMessage returns normally', () => {
```

Comment:
More test can be added to verify the formatting is correct. I'll wait until an agreed format is decided.